l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,286 @@
+            msg.sender,
+            salt,
+            L2ContractHelper.hashL2Bytecode(bytecodes.l2SharedBridgeProxyBytecode),
+            keccak256(l2SharedBridgeProxyConstructorData)
+        );
+
+        console.log("Computed L2 bridge proxy address:", proxyContractAddress);
+        console.log("L1 shared bridge proxy:", config.l1SharedBridgeProxy);
+        console.log("L1 ERC20 bridge proxy:", config.erc20BridgeProxy);
+        console.log("L2 governor addr:", l2GovernorAddress);
+
+        return proxyContractAddress;
+    }
+
+    function prepareRegisterHyperchainCall() internal view returns (IGovernance.Call memory) {
+        Bridgehub bridgehub = Bridgehub(config.bridgehub);
+
+        bytes memory data = abi.encodeCall(
+            bridgehub.createNewChain,
+            (
+                config.chainId,
+                config.stateTransitionProxy,
+                config.baseToken,
+                config.bridgehubCreateNewChainSalt,
+                config.chainAdmin,
+                config.diamondCutData
+            )
+        );
+
+        return IGovernance.Call({target: config.bridgehub, value: 0, data: data});
+    }
+
+    function prepareInitializeChainGovernanceCall(
+        address l2SharedBridgeProxy
+    ) internal view returns (IGovernance.Call memory) {
+        L1SharedBridge bridge = L1SharedBridge(config.l1SharedBridgeProxy);
+
+        bytes memory data = abi.encodeCall(bridge.initializeChainGovernance, (config.chainId, l2SharedBridgeProxy));
+
+        return IGovernance.Call({target: config.l1SharedBridgeProxy, value: 0, data: data});
+    }
+
+    function scheduleTransparentCalldata(
+        IGovernance.Call[] memory calls,
+        IGovernance.Call memory initChainGovCall
+    ) internal {
+        IGovernance governance = IGovernance(config.governance);
+
+        IGovernance.Operation memory operation = IGovernance.Operation({
+            calls: calls,
+            predecessor: bytes32(0),
+            salt: bytes32(config.bridgehubCreateNewChainSalt)
+        });
+
+        bytes memory scheduleCalldata = abi.encodeCall(governance.scheduleTransparent, (operation, 0));
+        bytes memory executeCalldata = abi.encodeCall(governance.execute, (operation));
+        console.log("Completed");
+
+        IGovernance.Call[] memory initChainGovArray = new IGovernance.Call[](1);
+        initChainGovArray[0] = initChainGovCall;
+
+        IGovernance.Operation memory operation2 = IGovernance.Operation({
+            calls: initChainGovArray,
+            predecessor: bytes32(0),
+            salt: bytes32(config.bridgehubCreateNewChainSalt)
+        });
+
+        bytes memory scheduleCalldata2 = abi.encodeCall(governance.scheduleTransparent, (operation2, 0));
+        bytes memory executeCalldata2 = abi.encodeCall(governance.execute, (operation2));
+
+        saveOutput(scheduleCalldata, executeCalldata, scheduleCalldata2, executeCalldata2);
+    }
+
+    function saveOutput(
```

Comment:
add comment - especially on what is saved and where.

Reply:
comment missing

Reply:
added